R/XcmsExperiment.R

```diff
@@ -1677,32 +1648,32 @@ setMethod(
         ## Populate with chrom peaks.
         nf <- nrow(fts)
         js <- seq_len(ncol(chrs))
-        pks_empty <- chromPeaks(object)[integer(), ]
-        pkd_empty <- chromPeakData(object)[integer(), ]
+        pks_empty <- .chromPeaks(object)[integer(), ]
+        pkd_empty <- as(.chromPeakData(object)[integer(), ], "DataFrame")
```

Comment:
What is happening here ?

Reply:
I created new internal functions `.chromPeaks` and `.chromPeakData` to extract the content of the `chromPeaks` (or `chromPeakData`) slot from an xcms result object. We have two xcms result objects now, `XCMSnExp` and `XcmsExperiment`. The former stores the data in a quite clumsy way, and extracting the data is a little more tricky. The latter is the new preferred result object. There is also the `chromPeaks` and `chromPeakData` **method** available for each of the two result objects, but these methods do quite some additional checks and allow subset/filter the content. The new internal functions (are not expected to be called by the user) are simple helpers to extract the full chromPeaks and chromPeakData content without any filtering etc. 

I could also have created a method for that, but I decided against that and just added a function with an if/else inside.

Then, for chromPeakData I have also to ensure that the result is returned as a `DataFrame`, thus I'm calling in addition the `as(..., "DataFrame"`. For `XCMSnExp` `.chromPeakData` will already return a `DataFrame`, so nothing will happen, while for `XcmsExperiment` I'm storing the data as a `data.frame`, thus I need to convert first.

Hope this explained it?